modules/service/src/main/scala/lucuma/odb/graphql/mapping/TargetEnvironmentMapping.scala

```diff
@@ -109,7 +121,28 @@ trait TargetEnvironmentMapping[F[_]: Temporal]
       (pid, oid, obsTime) =>
         services.use { s =>
           s.guideEnvironmentService(httpClient, itcClient, commitHash, plannedTimeCalculator)
-            .get(pid, oid, obsTime.toInstant)
+            .get(pid, oid, obsTime)
+            .map {
+              case Left(e)  => Result.failure(e.format)
+              case Right(s) => s.success
+            }
+        }
+
+    effectHandler("guideEnvironments", readEnv, calculate)
+  }
+
+  def guideAvailabilityQueryHandler: EffectHandler[F] = {
+    val readEnv: Env => Result[(Timestamp, Timestamp)] = env =>
+      for {
+        start <- env.getR[Timestamp](AvailabilityStartParam)
+        end   <- env.getR[Timestamp](AvailabilityEndParam)
+      } yield (start, end)
+
+    val calculate: (Program.Id, Observation.Id, (Timestamp, Timestamp)) => F[Result[List[GuideEnvironmentService.AvailabilityPeriod]]] =
+      (pid, oid, period) =>
+        services.use { s =>
+          s.guideEnvironmentService(httpClient, itcClient, commitHash, plannedTimeCalculator)
+            .availability(pid, oid, period._1, period._2)
```

Comment:
Good catch on both accounts. Maybe the `fieldName` parameter was used at one time? I will put in a new PR to remove it since it affects other handlers as well.